internal/clients/azure.go

```diff
@@ -69,33 +76,70 @@ func TerraformSetupBuilder(version, providerSource, providerVersion string) terr
 			return ps, errors.Wrap(err, errTrackUsage)
 		}
 
-		data, err := xpresource.CommonCredentialExtractor(ctx, pc.Spec.Credentials.Source, client, pc.Spec.Credentials.CommonCredentialSelectors)
-		if err != nil {
-			return ps, errors.Wrap(err, errExtractCredentials)
-		}
-		azureCreds := map[string]string{}
-		if err := json.Unmarshal(data, &azureCreds); err != nil {
-			return ps, errors.Wrap(err, errUnmarshalCredentials)
-		}
-
 		ps.Configuration = map[string]interface{}{
 			keyTerraformFeatures: struct{}{},
 			// Terraform AzureRM provider tries to register all resource providers
 			// in Azure just in case if the provider of the resource you're
 			// trying to create is not registered and the returned error is
-			// ambigious. However, this requires service principal to have provider
+			// ambiguous. However, this requires service principal to have provider
 			// registration permissions which are irrelevant in most contexts.
 			// For details, see https://github.com/crossplane-contrib/provider-jet-azure/issues/104
 			keySkipProviderRegistration: true,
-			keyTerraformSubscriptionID:  azureCreds[keyAzureSubscriptionID],
 		}
-		// set credentials environment
-		ps.Env = []string{
-			fmt.Sprintf(fmtEnvVar, envSubscriptionID, azureCreds[keyAzureSubscriptionID]),
-			fmt.Sprintf(fmtEnvVar, envTenantID, azureCreds[keyAzureTenantID]),
-			fmt.Sprintf(fmtEnvVar, envClientID, azureCreds[keyAzureClientID]),
-			fmt.Sprintf(fmtEnvVar, envClientSecret, azureCreds[keyAzureClientSecret]),
+
+		var err error
+		switch pc.Spec.Credentials.Source { //nolint:exhaustive
+		case xpv1.CredentialsSourceInjectedIdentity:
+			err = msiAuth(pc, &ps)
+		default:
+			err = spAuth(ctx, pc, &ps, client)
+		}
+
+		if pc.Spec.Credentials.Source == xpv1.CredentialsSourceInjectedIdentity {
+			ps.Configuration[keyUseMSI] = "true"
+			if pc.Spec.ClientID != nil && len(*pc.Spec.ClientID) != 0 {
+				ps.Configuration[keyClientID] = *pc.Spec.ClientID
+			}
+
 		}
```

Comment:
don't we cover this code block in `msiAuth` function?

Reply:
Oops :) Yes, correct, removed that block. Thank you!